examples/wCCD/src/wCCD.tsx

```diff
@@ -101,146 +112,143 @@ async function updateWCCDBalanceAccount(account: string, setAmountAccount: (x: b
+        if (activeConnector) {
+            setWaitingForUser(true);
+            activeConnector
+                .connect()
+                .then((c) => {
+                    console.log('setting wallet connection', c);
+                    setActiveConnection(c);
+                })
+                .catch(console.error)
+                .finally(() => setWaitingForUser(false));
+        }
+    }, [activeConnector]);
 
-export default function wCCD({ handleGetAccount, handleNotConnected }: Props) {
-    const { account, isConnected } = useContext(state);
     const [admin, setAdmin] = useState<string>();
-    const [isWrapping, setIsWrapping] = useState<boolean>(true);
-    const [hash, setHash] = useState<string>('');
-    const [error, setError] = useState<string>('');
-    const [flipped, setflipped] = useState<boolean>(false);
-    const [waitForUser, setWaitForUser] = useState<boolean>(false);
-    const [amountAccount, setAmountAccount] = useState<bigint>(0n);
-    const inputValue = useRef<HTMLInputElement>(null);
 
     useEffect(() => {
-        if (isConnected) {
-            viewAdmin(setAdmin);
+        // Update admin contract.
+        if (activeConnection) {
+            withJsonRpcClient(activeConnection, (rpcClient) => viewAdmin(rpcClient, setAdmin)).catch(console.error);
         }
+    }, [activeConnection]);
 
-        if (isConnected) {
-            if (account) {
-                updateWCCDBalanceAccount(account, setAmountAccount);
-            }
-        }
-    }, [isConnected]);
-
-    const handleOnClick = useCallback(() => {
-        setWaitForUser(true);
-
-        detectConcordiumProvider()
-            .then((provider) => provider.connect())
-            .then(handleGetAccount)
-            .then(() => {
-                setWaitForUser(false);
-                detectConcordiumProvider()
-                    // Check if the user is connected to testnet by checking if the testnet genesisBlock exists.
-                    // Throw a warning and disconnect if not. We only want to
-                    // allow users to interact with our testnet smart contracts.
-                    .then((provider) =>
-                        provider
-                            .getJsonRpcClient()
-                            .getCryptographicParameters(TESTNET_GENESIS_BLOCK_HASH.toString())
-                            .then((result) => {
-                                if (result === undefined || result?.value === null) {
-                                    handleNotConnected();
-                                    /* eslint-disable no-alert */
-                                    window.alert(
-                                        'Your JsonRpcClient in the Concordium browser wallet cannot connect. Check if your Concordium browser wallet is connected to testnet!'
-                                    );
-                                }
-                            })
-                    );
```

Comment:
Here.

Reply:
Yes? That check has been moved to `onChainChanged` in `WithWalletConnection`.